Refactor: tidy up code for making requests and remove placeholders

tjtanjin opened the pull request:
Refactor to tidy up code for making requests.
Remove placeholders - now properly supports making a basic get request.

Note:
- The use of the terminology `address` was used in this code addition, it will eventually be updated to `url` as the `Endpoint` attributes are updated.

Closes #128

tlylt reviewed (approved):
The `SendRequest` class perhaps can be more aptly named as `Request` given that we are modeling a `Request` and the word `Send` seems redundant. 
Just my 2 cents, other than that LGTM

tjtanjin commented:
> The `SendRequest` class perhaps can be more aptly named as `Request`? Given that we are indeed modeling a thing that is called `Request` and the word `Send` seems redundant.
> Just my 2 cents, other than that LGTM

Ok I renamed the logic classes under `endpoint` and renamed the `execute` method to `send` as well which sounds more apt: `new GetRequest(endpointToSend).send();`

Merging it since the changes are small :D
